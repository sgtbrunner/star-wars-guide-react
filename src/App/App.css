@font-face {
  font-family: 'Star Jedi';
  src: url(../assets/fonts/starjedi.ttf);
}

#temp-page {
  max-width: 100%;
  max-height: 100vh;
  height: 100%;
}

#main {
  background-image: url('../assets/img/bg.jpg');
  padding: 15px 0;
}

.page-loader {
  text-align: center;
  vertical-align: center;
  font-family: 'Star Jedi';
  font-size: 10vh;
  padding: 25vh 10vh;
}

@keyframes flickerAnimation {
  0% {
    opacity: 1;
  }
  50% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}
@-o-keyframes flickerAnimation {
  0% {
    opacity: 1;
  }
  50% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}
@-moz-keyframes flickerAnimation {
  0% {
    opacity: 1;
  }
  50% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}
@-webkit-keyframes flickerAnimation {
  0% {
    opacity: 1;
  }
  50% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}
.animate-flicker {
  -webkit-animation: flickerAnimation 1s infinite;
  -moz-animation: flickerAnimation 1s infinite;
  -o-animation: flickerAnimation 1s infinite;
  animation: flickerAnimation 1s infinite;
}

.white-backdrop {
  background-color: rgba(240, 240, 240);
  border-color: gray;
}

#main > .white-backdrop {
  min-height: 67vh;
  background-color: rgba(255, 255, 255, 0.8);
}

@keyframes spin {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
