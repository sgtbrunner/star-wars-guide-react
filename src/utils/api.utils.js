import { BASE_API_URL } from './constants.utils';

const fetchData = async ({ path, url, resource }) => {
  await fetch(url || `${BASE_API_URL}${path}`, {
    method: 'GET',
    headers: { 'Content-Type': 'application/json' },
  })
    .then((response) => response.json())
    .then(async (data) => {
      resource.push(...data.results);
      if (data.next) await fetchData({ url: data.next, resource });
    });
  return resource;
};

export default {
  getData: (path) => fetchData({ path, resource: [] }),
};
