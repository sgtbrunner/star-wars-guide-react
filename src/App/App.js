import React, { useEffect } from 'react';
import PropTypes from 'prop-types';

import Footer from '../components/Blocks/Footer';
import Header from '../components/Blocks/Header';
import Main from '../components/Blocks/Main';
import Loader from '../components/Loader';
import './App.css';

const App = ({
  isCharactersLoading,
  loadCharacters,
  loadFilms,
  loadPlanets,
  loadSpecies,
  loadStarships,
  loadVehicles,
}) => {
  useEffect(() => {
    const fetchData = async () => {
      await loadCharacters();
      loadFilms();
      loadPlanets();
      loadSpecies();
      loadStarships();
      loadVehicles();
    };

    fetchData();
  }, [loadCharacters, loadFilms, loadPlanets, loadSpecies, loadStarships, loadVehicles]);

  return isCharactersLoading ? (
    <Loader />
  ) : (
    <>
      <Header />
      <Main />
      <Footer />
    </>
  );
};

App.propTypes = {
  isCharactersLoading: PropTypes.bool.isRequired,
  loadCharacters: PropTypes.func.isRequired,
  loadFilms: PropTypes.func.isRequired,
  loadPlanets: PropTypes.func.isRequired,
  loadSpecies: PropTypes.func.isRequired,
  loadStarships: PropTypes.func.isRequired,
  loadVehicles: PropTypes.func.isRequired,
};

export default App;
