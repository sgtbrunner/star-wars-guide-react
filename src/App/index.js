import { connect } from 'react-redux';

import loadCharacters from '../redux/characters/characters.actions';
import loadFilms from '../redux/films/films.actions';
import loadPlanets from '../redux/planets/planets.actions';
import loadSpecies from '../redux/species/species.actions';
import loadStarships from '../redux/starships/starships.actions';
import loadVehicles from '../redux/vehicles/vehicles.actions';
import App from './App';

const mapStateToProps = (state) => ({
  isCharactersLoading: !state.characters.data.length,
  showDialog: state.dialog.isOpen,
});

const mapDispatchToProps = (dispatch) => ({
  loadCharacters: () => loadCharacters(dispatch),
  loadFilms: () => loadFilms(dispatch),
  loadPlanets: () => loadPlanets(dispatch),
  loadSpecies: () => loadSpecies(dispatch),
  loadStarships: () => loadStarships(dispatch),
  loadVehicles: () => loadVehicles(dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
