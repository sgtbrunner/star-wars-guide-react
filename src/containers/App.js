import React, {Component} from 'react';
import Header from '../components/Header';
import SearchBox from '../components/SearchBox';
import CardList from '../components/CardList';
import CreateList from '../components/SupportFunctions';
import Modal from '../components/Modal';
import {getStats} from '../components/SupportFunctions';
import {getFilms} from '../components/SupportFunctions';
import Footer from '../components/Footer';
import './App.css';

class App extends Component {
  // STATE DECLARATION AND INITIALIZATION
  constructor() {
    super()
    this.state = {
      characters: [],
      searchfield: '',
      clickedcard: '',
      race: '',
      planet: '',
      movies: '',
      showmodal: false
    }
  }

  // componentDidMount() IS A LIFECYCLE HOOK THAT RUNS ONCE THE APP COMPONENT IS MOUNTED IN 
  // INDEX.HTML. IN THIS CASE, IT UPDATES THIS.STATE.CHARACTERS WITH THE RETURNED ARRAY OF 
  // CHARACTERS FROM GETCHARACTERS()
  componentDidMount() {
  // ** Fetches all characters from SWAPI
    CreateList().then(response => this.setState({characters: response}));
  }

  // TRIGGERED EVERY TIME THE SEARCHBOX CHANGES
  // ** Actively changes displayed characters to seached value
  onSearchChange = (event) => {
    this.setState({searchfield: event.target.value});
  }

  // TRIGGERED EVERY TIME A CARD IS CLICKED
  openModal = (event) => {
  // ** Fetches asynchronous character information and open modal
    this.setState({clickedcard: this.state.characters[event.target.id]})
    getStats(this.state.characters[event.target.id].species)
    .then(response => this.setState({race: response}));
    getStats(this.state.characters[event.target.id].homeworld)
    .then(response => this.setState({planet: response}));
    getFilms(this.state.characters[event.target.id].films)
    .then(response => this.setState({movies: response}));
    this.setState({showmodal: true});
  }

  // TRIGGERED EVERYTIME X OR OUTER MODAL IS CLICKED
  onCloseClick = () => {
  // ** Set states to initial values and close modal
    this.setState({
      clickedcard: '',
      race: '',
      planet: '',
      movies: '',
      showmodal: false
    });
  }

  // RENDER TO INDEX.HTML
  render() {
    const filteredCharacters = this.state.characters.filter(character => {
    // Needs .toLowerCase() on both sides of the declaration to compare input to user.name on the same level
        return character.name.toLowerCase().includes(this.state.searchfield.toLowerCase())
      } 
    );

    if (!this.state.characters.length) {
        return (
            <div id="temp-page">
              <div className ="page-loader animate-flicker">Please wait...</div>
            </div>      
      )} else {
            return (
              <div>
                <Header />
                <div id="flex-container">
                  <SearchBox searchChange={this.onSearchChange}/>
                  <CardList 
                      characters = {this.state.characters}
                      filteredCharacters = {filteredCharacters}
                      openModal = {this.openModal}
                  />
                  <Modal 
                    showModal = {this.state.showmodal} 
                    character = {this.state.clickedcard}
                    race = {this.state.race}
                    planet = {this.state.planet}
                    movies = {this.state.movies}
                    onCloseClick = {this.onCloseClick}
                  />
                </div>
                <Footer />
              </div>
            )        
        }
  }
}

export default App;