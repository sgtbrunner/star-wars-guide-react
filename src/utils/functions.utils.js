import { BASE_IMAGE_URL } from './constants.utils';

export const getIdParamFromUrl = (url) => parseInt(url?.replace(/\D/g, ''), 10);

const mapArrayPropertyToId = (item) => getIdParamFromUrl(item);

export const capitalizeFirstLetter = (string) => string[0].toUpperCase() + string.slice(1);

export const getImageUrl = (index) =>
  typeof index === 'number' || (typeof index === 'string' && Number(index))
    ? `${BASE_IMAGE_URL}${Number(index) < 16 ? Number(index) + 1 : Number(index) + 2}.jpg`
    : null;

export const updateCharactersProperties = (data) =>
  data.map((character, index) => ({
    id: index,
    ...character,
    homeworld: getIdParamFromUrl(character.homeworld),
    films: character.films.map(mapArrayPropertyToId),
    species: character.species.map(mapArrayPropertyToId),
  }));
