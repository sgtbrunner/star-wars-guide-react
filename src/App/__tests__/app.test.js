import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import { longCharactersList, species, planets, films } from '../../utils/test-mock.utils';
import { DATA_INITIAL_STATE, DIALOG_INITIAL_STATE } from '../../utils/constants.utils';

import App from '../index';

const props = {
  isCharactersLoading: true,
  loadCharacters: jest.fn(),
  loadFilms: jest.fn(),
  loadPlanets: jest.fn(),
  loadSpecies: jest.fn(),
  showDialog: false,
};

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

const LOADING_STATE = {
  ...DATA_INITIAL_STATE,
  isLoading: true,
};

const initialData = {
  characters: LOADING_STATE,
  films: LOADING_STATE,
  planets: LOADING_STATE,
  species: LOADING_STATE,
  dialog: DIALOG_INITIAL_STATE,
};

const initialStore = mockStore(initialData);

const mountApp = (store = initialStore, newProps = {}) =>
  render(
    <Provider store={store}>
      <App {...props} {...newProps} />,
    </Provider>
  );

describe('App', () => {
  const loadedData = {
    ...initialData,
    characters: {
      ...DATA_INITIAL_STATE,
      data: longCharactersList,
    },
    films: {
      ...DATA_INITIAL_STATE,
      data: films,
    },
    planets: {
      ...DATA_INITIAL_STATE,
      data: planets,
    },
    species: {
      ...DATA_INITIAL_STATE,
      data: species,
    },
  };

  const openDialogData = {
    ...loadedData,
    dialog: {
      isOpen: true,
      characterId: 0,
    },
  };

  const loadedStore = mockStore(loadedData);
  const openDialogStore = mockStore(openDialogData);

  it('should match snapshot when characters are loading', () => {
    const wrapper = mountApp();

    expect(wrapper).toMatchSnapshot();
  });

  it('should render loader component while characters are loading', () => {
    mountApp();

    const loader = screen.getByText(/please wait/i);
    expect(loader).toBeInTheDocument();
  });

  it('should match snapshot when characters are loaded and no dialog is open', () => {
    const wrapper = mountApp(loadedStore, { isCharactersLoading: false });

    expect(wrapper).toMatchSnapshot();
  });

  it('should render header block component', () => {
    mountApp(loadedStore, { isCharactersLoading: false });

    const header = screen.getByAltText('logo');
    expect(header).toBeInTheDocument();
  });

  it('should render main block component', () => {
    mountApp(loadedStore, { isCharactersLoading: false });

    // asserts for rendered button cards
    const cards = screen.getAllByRole('button');
    expect(cards).toHaveLength(6);
    cards.forEach((card, index) => expect(card).toHaveTextContent(longCharactersList[index].name));
  });

  it('should render footer block component', () => {
    mountApp(loadedStore, { isCharactersLoading: false });

    const title = screen.getByText(/guilherme brunner/i);
    const link = screen.getByRole('link', { name: /visit my portfolio/i });

    expect(title).toBeInTheDocument();
    expect(link).toBeInTheDocument();
  });

  it('should match snapshot when characters are loaded and dialog is open', () => {
    const wrapper = mountApp(openDialogStore, { isCharactersLoading: false, showDialog: true });

    expect(wrapper).toMatchSnapshot();
  });

  it('should render dialog content when characters are loaded and dialog is open', () => {
    mountApp(openDialogStore, { isCharactersLoading: false, showDialog: true });

    const portrait = screen.getByRole('img', { name: '0' });
    const stats = screen.getAllByRole('listitem');

    expect(portrait).toBeInTheDocument();
    expect(stats).toHaveLength(7);
  });
});
