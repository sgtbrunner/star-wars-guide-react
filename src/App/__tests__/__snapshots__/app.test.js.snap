// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot when characters are loaded and dialog is open 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        href="#main"
      >
        <div
          class="header"
        >
          <img
            alt="logo"
            class="cover-image"
            src="logo.png"
          />
        </div>
      </a>
      <div
        id="main"
      >
        <div
          class="pa2 br3 searchbox"
        >
          <input
            class="f4 pa2 w-100 ba bw1 custom-field"
            placeholder="find your favorite Star Wars character..."
            type="search"
          />
        </div>
        <div
          class="cardlist"
        >
          <button
            class="card pointer"
            id="0"
            type="button"
          >
            <img
              alt="character-portrait"
              id="0"
              src="https://starwars-visualguide.com/assets/img/characters/1.jpg"
            />
            <div
              class="name"
              id="0"
            >
              Anakin Skywalker
            </div>
          </button>
          <button
            class="card pointer"
            id="1"
            type="button"
          >
            <img
              alt="character-portrait"
              id="1"
              src="https://starwars-visualguide.com/assets/img/characters/2.jpg"
            />
            <div
              class="name"
              id="1"
            >
              Obi-Wan Kenoby
            </div>
          </button>
          <button
            class="card pointer"
            id="2"
            type="button"
          >
            <img
              alt="character-portrait"
              id="2"
              src="https://starwars-visualguide.com/assets/img/characters/3.jpg"
            />
            <div
              class="name"
              id="2"
            >
              Yoda
            </div>
          </button>
          <button
            class="card pointer"
            id="3"
            type="button"
          >
            <img
              alt="character-portrait"
              id="3"
              src="https://starwars-visualguide.com/assets/img/characters/4.jpg"
            />
            <div
              class="name"
              id="3"
            >
              Darth Vader
            </div>
          </button>
          <button
            class="card pointer"
            id="4"
            type="button"
          >
            <img
              alt="character-portrait"
              id="4"
              src="https://starwars-visualguide.com/assets/img/characters/5.jpg"
            />
            <div
              class="name"
              id="4"
            >
              Luke Skywalker
            </div>
          </button>
          <button
            class="card pointer"
            id="5"
            type="button"
          >
            <img
              alt="character-portrait"
              id="5"
              src="https://starwars-visualguide.com/assets/img/characters/6.jpg"
            />
            <div
              class="name"
              id="5"
            >
              Palpatine
            </div>
          </button>
        </div>
      </div>
      <footer
        class="pa2 ph5-m ph6-l tc bg-near-black"
      >
        <small
          class="pa2 f6 db tc lightest-blue"
        >
          <b
            class="ttu"
          >
            GUILHERME BRUNNER
          </b>
           2021
        </small>
        <small
          class="pa2"
        >
          <a
            href="https://sgtbrunner.github.io/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <div
              class="f6 dib ph2 link lightest-blue dim"
            >
              Visit my 
              <b>
                PORTFOLIO
              </b>
            </div>
          </a>
        </small>
      </footer>
      <div
        class="overlay"
        id="overlay"
        role="dialog"
      >
        <div
          class="dialog-content"
        >
          <span
            class="close"
          >
            <button
              class="bn pointer bg-transparent"
              type="button"
            >
              Ã—
            </button>
          </span>
          <div
            class="stats-list"
          >
            <img
              alt="0"
              class="portrait"
              src="https://starwars-visualguide.com/assets/img/characters/1.jpg"
            />
            <div
              class="char-name"
            >
              Anakin Skywalker
            </div>
            <div
              class="stats"
              role="listitem"
            >
              <u>
                Species
              </u>
              : Human 
            </div>
            <div
              class="stats"
              role="listitem"
            >
              <u>
                Gender
              </u>
              : Male 
            </div>
            <div
              class="stats"
              role="listitem"
            >
              <u>
                Birth Year
              </u>
              : 41.9BBY 
            </div>
            <div
              class="stats"
              role="listitem"
            >
              <u>
                Height
              </u>
              : 188 cm
            </div>
            <div
              class="stats"
              role="listitem"
            >
              <u>
                Weight
              </u>
              : 84 Kg
            </div>
            <div
              class="stats"
              role="listitem"
            >
              <u>
                Homeworld
              </u>
              : Tatooine 
            </div>
            <div
              class="stats"
              role="listitem"
            >
              <u>
                Films
              </u>
              : The Phantom Menace, Attack of the Clones, Revenge of the Sith 
            </div>
          </div>
        </div>
      </div>
      ,
    </div>
  </body>,
  "container": <div>
    <a
      href="#main"
    >
      <div
        class="header"
      >
        <img
          alt="logo"
          class="cover-image"
          src="logo.png"
        />
      </div>
    </a>
    <div
      id="main"
    >
      <div
        class="pa2 br3 searchbox"
      >
        <input
          class="f4 pa2 w-100 ba bw1 custom-field"
          placeholder="find your favorite Star Wars character..."
          type="search"
        />
      </div>
      <div
        class="cardlist"
      >
        <button
          class="card pointer"
          id="0"
          type="button"
        >
          <img
            alt="character-portrait"
            id="0"
            src="https://starwars-visualguide.com/assets/img/characters/1.jpg"
          />
          <div
            class="name"
            id="0"
          >
            Anakin Skywalker
          </div>
        </button>
        <button
          class="card pointer"
          id="1"
          type="button"
        >
          <img
            alt="character-portrait"
            id="1"
            src="https://starwars-visualguide.com/assets/img/characters/2.jpg"
          />
          <div
            class="name"
            id="1"
          >
            Obi-Wan Kenoby
          </div>
        </button>
        <button
          class="card pointer"
          id="2"
          type="button"
        >
          <img
            alt="character-portrait"
            id="2"
            src="https://starwars-visualguide.com/assets/img/characters/3.jpg"
          />
          <div
            class="name"
            id="2"
          >
            Yoda
          </div>
        </button>
        <button
          class="card pointer"
          id="3"
          type="button"
        >
          <img
            alt="character-portrait"
            id="3"
            src="https://starwars-visualguide.com/assets/img/characters/4.jpg"
          />
          <div
            class="name"
            id="3"
          >
            Darth Vader
          </div>
        </button>
        <button
          class="card pointer"
          id="4"
          type="button"
        >
          <img
            alt="character-portrait"
            id="4"
            src="https://starwars-visualguide.com/assets/img/characters/5.jpg"
          />
          <div
            class="name"
            id="4"
          >
            Luke Skywalker
          </div>
        </button>
        <button
          class="card pointer"
          id="5"
          type="button"
        >
          <img
            alt="character-portrait"
            id="5"
            src="https://starwars-visualguide.com/assets/img/characters/6.jpg"
          />
          <div
            class="name"
            id="5"
          >
            Palpatine
          </div>
        </button>
      </div>
    </div>
    <footer
      class="pa2 ph5-m ph6-l tc bg-near-black"
    >
      <small
        class="pa2 f6 db tc lightest-blue"
      >
        <b
          class="ttu"
        >
          GUILHERME BRUNNER
        </b>
         2021
      </small>
      <small
        class="pa2"
      >
        <a
          href="https://sgtbrunner.github.io/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            class="f6 dib ph2 link lightest-blue dim"
          >
            Visit my 
            <b>
              PORTFOLIO
            </b>
          </div>
        </a>
      </small>
    </footer>
    <div
      class="overlay"
      id="overlay"
      role="dialog"
    >
      <div
        class="dialog-content"
      >
        <span
          class="close"
        >
          <button
            class="bn pointer bg-transparent"
            type="button"
          >
            Ã—
          </button>
        </span>
        <div
          class="stats-list"
        >
          <img
            alt="0"
            class="portrait"
            src="https://starwars-visualguide.com/assets/img/characters/1.jpg"
          />
          <div
            class="char-name"
          >
            Anakin Skywalker
          </div>
          <div
            class="stats"
            role="listitem"
          >
            <u>
              Species
            </u>
            : Human 
          </div>
          <div
            class="stats"
            role="listitem"
          >
            <u>
              Gender
            </u>
            : Male 
          </div>
          <div
            class="stats"
            role="listitem"
          >
            <u>
              Birth Year
            </u>
            : 41.9BBY 
          </div>
          <div
            class="stats"
            role="listitem"
          >
            <u>
              Height
            </u>
            : 188 cm
          </div>
          <div
            class="stats"
            role="listitem"
          >
            <u>
              Weight
            </u>
            : 84 Kg
          </div>
          <div
            class="stats"
            role="listitem"
          >
            <u>
              Homeworld
            </u>
            : Tatooine 
          </div>
          <div
            class="stats"
            role="listitem"
          >
            <u>
              Films
            </u>
            : The Phantom Menace, Attack of the Clones, Revenge of the Sith 
          </div>
        </div>
      </div>
    </div>
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`App should match snapshot when characters are loaded and no dialog is open 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        href="#main"
      >
        <div
          class="header"
        >
          <img
            alt="logo"
            class="cover-image"
            src="logo.png"
          />
        </div>
      </a>
      <div
        id="main"
      >
        <div
          class="pa2 br3 searchbox"
        >
          <input
            class="f4 pa2 w-100 ba bw1 custom-field"
            placeholder="find your favorite Star Wars character..."
            type="search"
          />
        </div>
        <div
          class="cardlist"
        >
          <button
            class="card pointer"
            id="0"
            type="button"
          >
            <img
              alt="character-portrait"
              id="0"
              src="https://starwars-visualguide.com/assets/img/characters/1.jpg"
            />
            <div
              class="name"
              id="0"
            >
              Anakin Skywalker
            </div>
          </button>
          <button
            class="card pointer"
            id="1"
            type="button"
          >
            <img
              alt="character-portrait"
              id="1"
              src="https://starwars-visualguide.com/assets/img/characters/2.jpg"
            />
            <div
              class="name"
              id="1"
            >
              Obi-Wan Kenoby
            </div>
          </button>
          <button
            class="card pointer"
            id="2"
            type="button"
          >
            <img
              alt="character-portrait"
              id="2"
              src="https://starwars-visualguide.com/assets/img/characters/3.jpg"
            />
            <div
              class="name"
              id="2"
            >
              Yoda
            </div>
          </button>
          <button
            class="card pointer"
            id="3"
            type="button"
          >
            <img
              alt="character-portrait"
              id="3"
              src="https://starwars-visualguide.com/assets/img/characters/4.jpg"
            />
            <div
              class="name"
              id="3"
            >
              Darth Vader
            </div>
          </button>
          <button
            class="card pointer"
            id="4"
            type="button"
          >
            <img
              alt="character-portrait"
              id="4"
              src="https://starwars-visualguide.com/assets/img/characters/5.jpg"
            />
            <div
              class="name"
              id="4"
            >
              Luke Skywalker
            </div>
          </button>
          <button
            class="card pointer"
            id="5"
            type="button"
          >
            <img
              alt="character-portrait"
              id="5"
              src="https://starwars-visualguide.com/assets/img/characters/6.jpg"
            />
            <div
              class="name"
              id="5"
            >
              Palpatine
            </div>
          </button>
        </div>
      </div>
      <footer
        class="pa2 ph5-m ph6-l tc bg-near-black"
      >
        <small
          class="pa2 f6 db tc lightest-blue"
        >
          <b
            class="ttu"
          >
            GUILHERME BRUNNER
          </b>
           2021
        </small>
        <small
          class="pa2"
        >
          <a
            href="https://sgtbrunner.github.io/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <div
              class="f6 dib ph2 link lightest-blue dim"
            >
              Visit my 
              <b>
                PORTFOLIO
              </b>
            </div>
          </a>
        </small>
      </footer>
      ,
    </div>
  </body>,
  "container": <div>
    <a
      href="#main"
    >
      <div
        class="header"
      >
        <img
          alt="logo"
          class="cover-image"
          src="logo.png"
        />
      </div>
    </a>
    <div
      id="main"
    >
      <div
        class="pa2 br3 searchbox"
      >
        <input
          class="f4 pa2 w-100 ba bw1 custom-field"
          placeholder="find your favorite Star Wars character..."
          type="search"
        />
      </div>
      <div
        class="cardlist"
      >
        <button
          class="card pointer"
          id="0"
          type="button"
        >
          <img
            alt="character-portrait"
            id="0"
            src="https://starwars-visualguide.com/assets/img/characters/1.jpg"
          />
          <div
            class="name"
            id="0"
          >
            Anakin Skywalker
          </div>
        </button>
        <button
          class="card pointer"
          id="1"
          type="button"
        >
          <img
            alt="character-portrait"
            id="1"
            src="https://starwars-visualguide.com/assets/img/characters/2.jpg"
          />
          <div
            class="name"
            id="1"
          >
            Obi-Wan Kenoby
          </div>
        </button>
        <button
          class="card pointer"
          id="2"
          type="button"
        >
          <img
            alt="character-portrait"
            id="2"
            src="https://starwars-visualguide.com/assets/img/characters/3.jpg"
          />
          <div
            class="name"
            id="2"
          >
            Yoda
          </div>
        </button>
        <button
          class="card pointer"
          id="3"
          type="button"
        >
          <img
            alt="character-portrait"
            id="3"
            src="https://starwars-visualguide.com/assets/img/characters/4.jpg"
          />
          <div
            class="name"
            id="3"
          >
            Darth Vader
          </div>
        </button>
        <button
          class="card pointer"
          id="4"
          type="button"
        >
          <img
            alt="character-portrait"
            id="4"
            src="https://starwars-visualguide.com/assets/img/characters/5.jpg"
          />
          <div
            class="name"
            id="4"
          >
            Luke Skywalker
          </div>
        </button>
        <button
          class="card pointer"
          id="5"
          type="button"
        >
          <img
            alt="character-portrait"
            id="5"
            src="https://starwars-visualguide.com/assets/img/characters/6.jpg"
          />
          <div
            class="name"
            id="5"
          >
            Palpatine
          </div>
        </button>
      </div>
    </div>
    <footer
      class="pa2 ph5-m ph6-l tc bg-near-black"
    >
      <small
        class="pa2 f6 db tc lightest-blue"
      >
        <b
          class="ttu"
        >
          GUILHERME BRUNNER
        </b>
         2021
      </small>
      <small
        class="pa2"
      >
        <a
          href="https://sgtbrunner.github.io/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            class="f6 dib ph2 link lightest-blue dim"
          >
            Visit my 
            <b>
              PORTFOLIO
            </b>
          </div>
        </a>
      </small>
    </footer>
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`App should match snapshot when characters are loading 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="loader"
      >
        <div
          class="loader-content animate-flicker"
        >
          Please wait...
        </div>
      </div>
      ,
    </div>
  </body>,
  "container": <div>
    <div
      class="loader"
    >
      <div
        class="loader-content animate-flicker"
      >
        Please wait...
      </div>
    </div>
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
